lib/puppet_x/puppetlabs/cd4pe_job_helper.rb

```diff
@@ -0,0 +1,163 @@
+require 'puppet_x'
+require 'json'
+
+module PuppetX::Puppetlabs
+  # A class to help with the job creation / running process for CD4PE
+  class CD4PEJobHelper < Object
+    attr_reader :docker_run_args
+
+    MANIFEST_TYPE = { 
+      :JOB => "JOB", 
+      :AFTER_JOB_SUCCESS => "AFTER_JOB_SUCCESS", 
+      :AFTER_JOB_FAILURE => "AFTER_JOB_FAILURE" }
+
+    def initialize(working_dir:, docker_image: nil, docker_run_args: nil)
+      @docker_image = docker_image
+      @docker_run_args = docker_run_args
+      @docker_based_job = !docker_image.nil?
+      @working_dir = working_dir
+
+      @local_jobs_dir = File.join(@working_dir, "cd4pe_job", "jobs", "unix")
+      @local_repo_dir = File.join(@working_dir, "cd4pe_job", "repo")
+      
+      @docker_run_args = docker_run_args.nil? ? '' : docker_run_args.join(' ')
+    end
+
+    def set_job_env_vars(task_params)
+      ENV["WEB_UI_ENDPOINT"] = task_params['cd4pe_web_ui_endpoint']
+      ENV["JOB_TOKEN"] = task_params['cd4pe_token']
+      ENV["JOB_OWNER"] = task_params['cd4pe_job_owner']
+      ENV["JOB_INSTANCE_ID"] = task_params['job_instance_id']
+    
+      user_specified_env_vars = task_params['env_vars']
+      if (!user_specified_env_vars.nil?)
+        user_specified_env_vars.each do |var|
+          pair = var.split("=")
+          key = pair[0]
+          value = pair[1]
+          ENV[key] = value
+      end
+    end
+
+    def make_working_dir(working_dir)
+      Dir.mkdir(working_dir) unless File.exists?(working_dir)
+    end
+
+    def get_job_script_and_control_repo
+      target_file = File.join(@working_dir, "cd4pe_job.tar.gz")
```

Comment:
are there any worries around overwriting existing?

Reply:
nope. @working_dir is a temp dir created by the caller, and if the caller is a bolt task (which it is in our case), the entire environment is brand new every time.